src/main/java/gyro/google/compute/HealthCheckResource.java

```diff
@@ -0,0 +1,475 @@
+    private HealthCheckSslHealthCheck sslHealthCheck;
+    private HealthCheckTcpHealthCheck tcpHealthCheck;
+    private Integer healthyThreshold;
+    private String name;
+    private String selfLink;
+    private Integer timeoutSec;
+    private String type;
+    private Integer unhealthyThreshold;
+
+    /**
+     * The name of the health check. (Required)
+     */
+    @Id
+    @Required
+    @Regex("(?:(?:[-a-z0-9]{1,63}\\.)*(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?):)?(?:[0-9]{1,19}|(?:[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?))")
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The description of the health check.
+     */
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The time (in seconds) for how often to send a health check.
+     */
+    @Updatable
+    public Integer getCheckIntervalSec() {
+        return checkIntervalSec;
+    }
+
+    public void setCheckIntervalSec(Integer checkIntervalSec) {
+        this.checkIntervalSec = checkIntervalSec;
+    }
+
+    /**
+     * The http Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckHttpHealthCheck
+     */
+    @Updatable
+    public HealthCheckHttpHealthCheck getHttpHealthCheck() {
+        return httpHealthCheck;
+    }
+
+    public void setHttpHealthCheck(HealthCheckHttpHealthCheck httpHealthCheck) {
+        this.httpHealthCheck = httpHealthCheck;
+    }
+
+    /**
+     * The https Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckHttpsHealthCheck
+     */
+    @Updatable
+    public HealthCheckHttpsHealthCheck getHttpsHealthCheck() {
+        return httpsHealthCheck;
+    }
+
+    public void setHttpsHealthCheck(HealthCheckHttpsHealthCheck httpsHealthCheck) {
+        this.httpsHealthCheck = httpsHealthCheck;
+    }
+
+    /**
+     * The http2 Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckHttp2HealthCheck
+     */
+    @Updatable
+    public HealthCheckHttp2HealthCheck getHttp2HealthCheck() {
+        return http2HealthCheck;
+    }
+
+    public void setHttp2HealthCheck(HealthCheckHttp2HealthCheck http2HealthCheck) {
+        this.http2HealthCheck = http2HealthCheck;
+    }
+
+    /**
+     * The ssl Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckSslHealthCheck
+     */
+    @Updatable
+    public HealthCheckSslHealthCheck getSslHealthCheck() {
+        return sslHealthCheck;
+    }
+
+    public void setSslHealthCheck(HealthCheckSslHealthCheck sslHealthCheck) {
+        this.sslHealthCheck = sslHealthCheck;
+    }
+
+    /**
+     * The tcp Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckTcpHealthCheck
+     */
+    @Updatable
```

Comment:
You should add a `@ConflictsWith` annotation to this because only one can be set at a time:
```suggestion
    @Updatable
    @ConflictsWith({"http-health-check", "https-health-check", "http2-health-check", "ssl-health-check"})
```